frontend/src/viewer/components/path-element.ts

```diff
@@ -332,13 +355,116 @@ export class PathCtrlElement extends connect(store)(LitElement) {
     }
   }
 
+  protected launchScoring(): void {
+    if (this.tracks && this.currentTrack !== null && this.tracks[this.currentTrack]) {
+      let fixes: BRecord[] = [];
+      for (let i = 0; i < this.tracks[this.currentTrack].fixes.lat.length; i++)
+        // Keep this to the bare minimum needed for igc-xc-score
+        fixes[i] = {
+          timestamp: this.tracks[this.currentTrack].fixes.ts[i],
+          latitude: this.tracks[this.currentTrack].fixes.lat[i],
+          longitude: this.tracks[this.currentTrack].fixes.lon[i],
+          pressureAltitude: this.tracks[this.currentTrack].fixes.alt[i],
+          gpsAltitude: this.tracks[this.currentTrack].fixes.alt[i],
+          valid: true,
+          extensions: {} as RecordExtensions,
+          fixAccuracy: null,
+          time: '',
+          enl: null
+        };
+      if (this.worker !== null)
+        this.worker.terminate();
+      this.worker = new Worker('js/xc-score-worker.js');
```

Comment:
Do we need to create a new Worker each time ?

Reply:
It is not that expensive, this is only when the user clicks the button. terminate allows the user to relaunch a new optimization even when the previous hasn't finished. Interrupting a running optimization and reusing the thread would add lots of complexity for a very marginal performance gain - currently once an optimization is launched, the thread won't yield the CPU until it is finished.

Reply:
Ok, makes sense.
If the worked is not re-used then we can release it when it's done (`this.worker = null`).